### PR TITLE
Match fp16 loss scaling defaults with NeMo2

### PR DESCRIPTION
**Pull request by ananthsub**
Found while investigating https://github.com/NVIDIA-NeMo/Megatron-Bridge/issues/353

Our current defaults don't match MCore/NeMo2:

https://github.com/NVIDIA/NeMo/blob/379b87679df2fb44509687b18be842ace70f0718/nemo/lightning/pytorch/plugins/mixed_precision.py#L119-L122

https://github.com/NVIDIA/Megatron-LM/blob/75e2efd10fd324819a29c5ce581fdf0f3c37b29f/megatron/core/optimizer/optimizer_config.py#L81-L99

carry forward from https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/354

### CODE REVIEW

File: src/megatron/bridge/training/mixed_precision.py [hunk 1]
```diff
@@ -53,10 +53,10 @@ class MixedPrecisionConfig:
     fp8_param_gather: bool = False
     # FP16 Loss scaling
     loss_scale: Optional[float] = None
-    initial_loss_scale: Optional[float] = None
-    min_loss_scale: Optional[float] = None
-    loss_scale_window: Optional[float] = None
-    hysteresis: Optional[float] = None
+    initial_loss_scale: Optional[float] = 4294967296  # 2^32
```
Comment (ananthsub) [inline src/megatron/bridge/training/mixed_precision.py]:
```suggestion
    initial_loss_scale: Optional[float] = 4294967296  # 2**32
```

**Review by maanug-nv — APPROVED** [review]
